File: teoria-de-numeros/factorial/factorial.cpp
#include <bits/stdc++.h>
using namespace std;
#define sz(arr) ((int) arr.size())
typedef long long ll;

//Forma 1
int fact[100];
void calculateFact() {
    fact[0] = 1;
    for (int i=1; i<100; i++) fact[i] = i * fact[i-1];
    return;
}

//Forma 2
int fact[100];
const int MOD = 10007;
void calculateFact() {
    fact[0] = 1;
    for (int i=1; i<100; i++) (fact[i] = i * fact[i-1]) % MOD;
    return;
}

int main() {
  ios::sync_with_stdio(false);
  cin.tie(0);
  return 0;
}

File: teoria-de-numeros/factorial/factorial.py
# Autor: Mateo Álvarez Murillo
# Fecha de creación: 2023

# Este código se proporciona bajo la Licencia MIT.
# Para más información, consulta el archivo LICENSE en la raíz del repositorio.

# Forma 1
fact = [0] * 100


def calculateFact() -> None:
    fact[0] = 1
    for i in range(1, 100):
        fact[i] = i * fact[i - 1]


# Forma 2
fact = [0] * 100
MOD = 10007


def calculateFact() -> None:
    fact[0] = 1
    for i in range(1, 100):
        fact[i] = (i * fact[i - 1]) % MOD
